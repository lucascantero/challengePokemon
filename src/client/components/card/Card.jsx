import React from 'react'
import { Link } from 'react-router-dom';

const Card = ({ dato }) => {

    return (
        <li className="card__li">
            <Link className="card__link" to={`/pokemon/${dato.id}`}>
                <div className="card__wrapper">               
                    <img src={dato.img} alt="pokemon" className="card__img" />
                    <h1 className="card__name">
                        {dato.name}
                    </h1>
                </div>
            </Link>
        </li>
    )
}

export default Card;