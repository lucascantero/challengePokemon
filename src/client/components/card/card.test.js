import React from 'react';
import Card from './Card.jsx';
import { act } from 'react-dom/test-utils';
import {render,unmountComponentAtNode} from 'react-dom';
import { MemoryRouter } from 'react-router-dom';

describe('Testing Card', () => {

    let container = null;

    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('Should match snapshot', () => {
        
        let cardDate = {
            "id": "id",
            "name" :"name",
            "img": "img",
        }

        act(() => {
            render(<MemoryRouter> <Card dato={cardDate}/> </MemoryRouter>, container)
        })

        expect(container.innerHTML).toMatchSnapshot();
    })

})