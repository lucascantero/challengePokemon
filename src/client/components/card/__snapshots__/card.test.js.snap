// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card Should match snapshot 1`] = `" <li class=\\"card__li\\"><a class=\\"card__link\\" href=\\"/pokemon/id\\"><div class=\\"card__wrapper\\"><img src=\\"img\\" alt=\\"pokemon\\" class=\\"card__img\\"><h1 class=\\"card__name\\">name</h1></div></a></li> "`;
